pages/api/login/index.ts

```diff
@@ -0,0 +1,4 @@
+import loginApi from '@framework/api/endpoints/login'
+import commerce from '@lib/api/commerce'
+
+export default loginApi(commerce)
```

Comment:
Why was this page changed?

Reply:
It happened as part of this commit: [`c3713ec` (#548)](https://github.com/vercel/commerce/pull/548/commits/c3713ec6e23f1b423a071a31221069995d419486). It's really just a rename from `login.ts --> login/index.ts`

Previously, there was a `pages/api/login.ts` file but I needed this API route to be dynamic, so I created a directory and moved:
```
pages/api/login.ts --> pages/api/login/index.ts
```
and added
```
pages/api/login/[...param].ts
```

I could have left it as `pages/api/login.ts` but I would have still created the directory for the dynamic route, so I thought it neater to put both routes within the directory (as `index.ts` and `[...param].ts`)

Reply:
I'm not quite sure of adding this to the core, can you explain a little more on why this extra route is needed?

Reply:
**Background:**
Commerce.js only supports passwordless authentication - the user will enter their email address, and an email will be sent to them containing a "magic link" which will automatically authenticate them. This link's URL will contain a token, and when clicked, the link will take the user to an API that can exchange this token for a JWT and finally redirect the user to the application along with their JWT. 

**For example:**
1. User enters their email address and submits login form
2. User receives an email with link: `https://commercejs.vercal.store/api/login/{TOKEN}`
3. An API at route `/api/login/{TOKEN}` is called and exchanges `TOKEN` for a JWT, and redirects the user back to `https://commercejs.vercal.store` with JWT

**Why things needed to change:**
This authentication flow requires a login endpoint that accepts a dynamic path i.e. not just `/login` but `/login/{TOKEN}`.

The existing login API on its own will not satisfy this, because a request to `/login/123` will not match an API route at `/login`. This is why I created the new `/login/[...param]` API.

Let me know if any of that doesn't make sense!

Reply:
Thanks for the heads up @cpv123 ! In that case I think you can add a rewrite on your providers next.config.js, something like this:

```js
const commerce = require('./commerce.config.json')

module.exports = {
  commerce,
  images: {
    domains: ['cdn.chec.io', 'ocdevops.blob.core.windows.net'],
  },
  rewrites() {
    return [{
      source: '/api/login/:token',
      destination: '/api/login?token=:token'
    }]
  }
}
```
And then on your `api/endpoints/login/login` you will receive a query param called token, that fits your needs?

Reply:
Amazing, thanks @goncy. 

I've reverted the original commit and instead used a rewrite as you've suggested.